src/crystal/system/win32/event_loop_iocp.cr

```diff
@@ -1,14 +1,22 @@
 require "crystal/system/print_error"
 
 module Crystal::EventLoop
-  @@queue = Deque(Fiber).new
+  @@queue = Deque(Event).new
 
   # Runs the event loop.
   def self.run_once : Nil
-    next_fiber = @@queue.pop?
+    next_event = @@queue.min_by { |e| e.wake_at }
```

Comment:
I guess this is fine for now, but wouldn't it be better to have a more specialized data structure, to avoid linearly check events at each iteration?

Reply:
Definitely! There are already a couple of comments on this in the original discussion at https://github.com/crystal-lang/crystal/pull/9957#discussion_r530624631

But this is fine for initial implementation. Performance improvements can follow later.